R/data.table.R

```diff
@@ -1937,12 +1939,14 @@ as.matrix.data.table <- function(x, rownames=NULL, rownames.value=NULL, ...) {
       else { # rownames is a column number already
         rownames <- as.integer(rownames)
         if (is.na(rownames) || rownames<1L || rownames>ncol(x))
-          stop(sprintf("as.integer(rownames)==%d which is outside the column number range [1,ncol=%d].", rownames, ncol(x)))
+          stop("as.integer(rownames)==", rownames,
```

Comment:
above comment `# rownames is a column number already` is confusing, could be improved :)
`rownames` is always scalar in this branch?